src/plugins/saved_objects_management/common/types/v1.ts

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { SavedObjectError } from '@kbn/core/types';
+import type { SavedObjectsNamespaceType } from '@kbn/core/public';
+
+/** Domain interfaces */
+
+/**
+ * Saved Object Management metadata associated with a saved object. See
+ * {@link SavedObjectWithMetadata}.
+ */
+export interface SavedObjectMetadata {
```

Comment:
Are the types that don't have the `V1` suffix not expected to change at all? I'd expect to see all the types and interfaces declared in this file to _only_ be the ones that change.

Shared types/interfaces that we can guarantee will _never_ change could like elsewhere, for example, in the main `types` file.

Keeping the versioned-types file as small as possible helps tracking changes over time.